docs/canvas/canvas-expression-lifecycle.asciidoc

```diff
@@ -0,0 +1,241 @@
+
+
+image::images/canvas-functions-can-take-arguments-donut-chart.png[Donut Chart]
+[source,text]
+----
+filters
+| demodata
+| pointseries color="state" size="max(price)"
+| pie hole=50
+| render
+----
+
+
+[[canvas-aliases-and-unnamed-arguments]]
+=== Aliases and Unnamed Arguments
+
+Argument definitions have one canonical name, which is always provided in the underlying code, but when used in an expression they often times include aliases that make them easier or faster to type, usually both. 
+
+Take, as one example, the <<mapColumn_fn>> function, which has 2 arguments: `expression`, the expression to use to produce a calculated value, and `name`, the name of column to place the resulting data in. The `expression` argument includes some aliases, namely `exp`, `fn`, and `function`. That means that you can use any of those four options to provide that argument’s value. So in your expression, if you use `mapColumn name=newColumn fn={string example}`, this produces the same output as if you wrote the more verbose `mapColumn name=newColumn expression={string example}`.
+
+There’s also a special type of alias which allows you to leave off the argument’s name entirely. The alias for this is simply `+_+`, which indicates that the argument is an _unnamed_ argument, and can be provided without explicitly naming it in the expression. The `name` argument here uses the _unnamed_ alias, which means that you can further simplify our example to `mapColumn newColumn fn={string example}`.
+
+NOTE: There can only be one _unnamed_ argument for each function.
+
+
+[[canvas-change-your-expression-change-your-output]]
+=== Change Your Expression, Change Your Output
+In the previous section, you saw how you can add arguments to a function to change its output, but you can also substitute one function for another in many situations. For example, you could change the visualization by swapping out the <<pie_fn, pie>> function for another *renderer*, a function that returns a `render` data type. 
+
+Let’s change that last pie visualization into a bubble chart by replacing the <<pie_fn, pie>> function with the <<plot_fn, plot>> function. This is possible because both functions can accept a `pointseries` data type as their _context_. Just switching the functions will work, but it won’t produce a useful visualization on its own, since you don’t have any axis dimensions, i.e. `x` or `y`, defined. You’ll also need to modify the <<pointseries_fn, pointseries>> function to change its output as well. In this case, you can change the `size` argument to `y`, so the maximum price values are plotted on the y-axis, and add an `x` argument using the `@timestamp` field in the data to plot those values over time. This leaves you with the following expression and produces a bubble chart showing the max price of each state over time:
```

Comment:
I would change the third sentence to:

Switching the functions will work, but it won't product a useful visualization on its own since you don't have the x-axis and y-axis defined.